src/app-layout/__tests__/trigger-button.test.tsx

```diff
@@ -0,0 +1,247 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+/* eslint simple-import-sort/imports: 0 */
+import React from 'react';
+import { ButtonProps } from '../../../lib/components/button';
+import { IconProps } from '../../../lib/components/icon/interfaces.js';
+import { KeyCode } from '@cloudscape-design/test-utils-core/utils';
+import TriggerButton, { TriggerButtonProps } from '../../../lib/components/app-layout/visual-refresh/trigger-button';
+import triggerButtonStyles from '../../../lib/components/app-layout/visual-refresh/styles.css.js';
+import tooltipStyles from '../../../lib/components/internal/components/tooltip/styles.selectors.js';
+
+import { act, render, fireEvent } from '@testing-library/react';
+import createWrapper from '../../../lib/components/test-utils/dom';
+
+const mockDrawerId = 'mock-drawer-id';
+const mockTestId = `awsui-app-layout-trigger-${mockDrawerId}`;
+const mockProps = {
+  ariaLabel: 'Aria label',
+  className: 'class-from-props',
+  iconName: 'bug',
+  testId: mockTestId,
+};
+const mockOtherEl = {
+  class: 'other-el-class',
+  text: 'other-element',
+};
+
+function delay() {
+  return act(() => new Promise(resolve => setTimeout(resolve)));
+}
+
+async function renderTriggerButton(props: Partial<TriggerButtonProps> = {}, ref: React.Ref<ButtonProps.Ref>) {
+  const renderProps = { ...mockProps, ...props };
+  const { container, rerender, getByTestId, getByText } = render(
+    <div>
+      <TriggerButton {...(renderProps as TriggerButtonProps)} ref={ref} />
+      <span className={mockOtherEl.class}>{mockOtherEl.text}</span>
```

Comment:
Note to self, I need to add one more test that clicks on the outer element here to confirm tooltip closes